include/AST/Types/FunctionTy.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef GLU_AST_TYPES_FUNCTIONTY_HPP
+#define GLU_AST_TYPES_FUNCTIONTY_HPP
+
+#include "TypeBase.hpp"
+#include <vector>
+
+namespace glu::types {
+
+/// @brief FunctionTy is a class that represents a function type in the AST.
+class FunctionTy : public TypeBase {
+private:
```

Comment:
Not needed it is implicit in class, we didn't explicitly declared it in other classes

Reply:
no defined rules for that at the moment, we can discuss it :)